### PR TITLE
Enhance bulk import scripts

### PR DESCRIPTION
**Pull request by quangbuule**
Enhancement:
- Import posts' like_count from post_thanks hack
- Embed YouTube in posts
- Using Set when importing topic_allowed_users, prevent duplicating user_id in same topic (occurred in my case).

Fix:
- Bug normalizing twice when importing private topic title

### CODE REVIEW

File: script/bulk_import/base.rb [hunk 1]
```diff
@@ -484,7 +486,8 @@ def create_records(rows, name, columns)
         mapped = yield(row)
         next unless mapped
         processed = send(process_method_name, mapped)
-        imported_ids << mapped[:imported_id]
+        imported_ids << mapped[:imported_id] unless mapped[:imported_id].nil?
+        imported_ids |= mapped[:imported_ids] unless mapped[:imported_ids].nil?
```
Comment (ZogStriP) [inline script/bulk_import/base.rb]: Where is `mapped[:imported_ids]` set?

File: script/bulk_import/base.rb [hunk 2]
```diff
@@ -538,7 +541,18 @@ def random_email
   end
 
   def pre_cook(raw)
-    cooked = @markdown.render(raw).scrub.strip
+    cooked = raw
+
+    # Convert YouTube URLs to lazyYT DOMs before being transformed into links
+    cooked.gsub!(/\nhttps\:\/\/www.youtube.com\/watch\?v=(\w+)\n/m) do
```
Comment (ZogStriP) [inline script/bulk_import/base.rb]: Why do you need the `/m` modifier on that regexp?

File: script/bulk_import/vbulletin.rb [hunk 3]
```diff
@@ -490,5 +495,3 @@ def mysql_query(sql)
   end
 
 end
-
-BulkImport::VBulletin.new.run
```
Comment (ZogStriP) [inline script/bulk_import/vbulletin.rb]: Why was that line deleted?

File: script/bulk_import/vbulletin.rb [hunk 4]
```diff
@@ -396,11 +402,10 @@ def import_private_topics
 
       next if @imported_topics.has_key?(key)
       @imported_topics[key] = row[0] + PRIVATE_OFFSET
-
       {
         archetype: Archetype.private_message,
         imported_id: row[0] + PRIVATE_OFFSET,
-        title: normalize_text(title),
+        title: title,
```
Comment (ZogStriP) [inline script/bulk_import/vbulletin.rb]: 👍

File: script/bulk_import/vbulletin.rb [hunk 5]
```diff
@@ -410,7 +415,7 @@ def import_private_topics
   def import_topic_allowed_users
     puts "Importing topic allowed users..."
 
-    allowed_users = []
+    allowed_users = Set.new
```
Comment (ZogStriP) [inline script/bulk_import/vbulletin.rb]: 👍

File: script/bulk_import/vbulletin.rb (hunk 3 above)
Comment (quangbuule) [inline script/bulk_import/vbulletin.rb]: That line was deleted by mistake. I'll restore the line.

File: script/bulk_import/base.rb (hunk 2 above)
Comment (quangbuule) [inline script/bulk_import/base.rb]: Some posts contain several youtube links, what do you think?

Comment (ZogStriP) [inline script/bulk_import/base.rb]: The `/m` modifier is for "multiline" ;)

The `gsub` is taking care of the "several youtube links"

File: script/bulk_import/base.rb (hunk 1 above)
Comment (quangbuule) [inline script/bulk_import/base.rb]: This is for derived classes.

I am currently extends `BulkImport::VBulletin` for my case and think that imported_ids may be useful for future works.

I combined some old categories into new categories. I am having .yml file that have a list of new categories and imported_ids which are ids of old categories.

File: script/bulk_import/base.rb (hunk 2 above)
Comment (quangbuule) [inline script/bulk_import/base.rb]: Ahh, great to hear, thanks for that 😄 ! thought m stands for multiple 😛

File: script/bulk_import/base.rb (hunk 1 above)
Comment (quangbuule) [inline script/bulk_import/base.rb]: What do you say?
I pushed the fix of the rest btw.

Comment (quangbuule) [inline script/bulk_import/base.rb]: I mean multiple old categories into one new category. For clarity, here is a sample of my approach, which made my result very neat:
```yml
categories:
  - id: 1
    pre_existed: true
    is_uncategorized: true
  - id: 3
    pre_existed: true
    imported_ids: [4]
  - id: 4
    imported_ids: [27, 52]
    pre_existed: true
  - id: 5
    name: YAMAHA
    description: Discussion about YAMAHA keyboards.
    imported_ids: [76, 77, 78]
    children_categories:
      - id: 6
        name: External data
        description: YAMAHA keyboards' external data.
        imported_ids: [6, 7, 8, 46]
```

**Comment by ZogStriP** [comment]
Thanks 👍
